feat: add maxDisplayTags with interactive dropdown management

sanju9645 opened the pull request:
## 🎯 Overview
This PR adds a new `maxDisplayTags` configuration option that allows users to limit the number of visible selected tags in the container. When more tags are selected than the limit, a clickable "+ X more" indicator is shown, which opens an interactive dropdown for managing all selected tags.

## ✨ New Features

### maxDisplayTags Configuration
- **New config option:** `maxDisplayTags` to control visible tag count
- **Default behavior:** Shows all tags when not specified (backward compatible)
- **Flexible usage:** Can be set to any positive number

### Interactive "+ X more" Indicator
- **Clickable indicator:** Shows count of hidden tags
- **Professional styling:** Gray background with hover effects
- **Consistent design:** Matches existing UI patterns

### Tag Management Dropdown
- **View all tags:** Click "+ X more" to see all selected tags
- **Individual removal:** Click × button to remove specific tags
- **Auto-hide:** Dropdown closes when no tags remain
- **Consistent UX:** Same styling as main dropdown

## �� UI/UX Improvements
- **Clean interface:** Prevents UI clutter with many selections
- **Smooth interactions:** Hover effects and transitions
- **Professional appearance:** Consistent with existing design
- **Intuitive controls:** Clear visual feedback for interactions

## 🔧 Technical Implementation
- **New dropdown element:** `selectedTagsDropdown` for tag management
- **Enhanced rendering:** `renderSelectedTags()` with display limit logic
- **Event handling:** Proper click and hover event management
- **CSS styling:** Comprehensive styles for new components

## �� Documentation
- **Updated README:** Detailed configuration examples
- **Usage instructions:** Clear examples for different scenarios
- **Feature documentation:** Complete guide to interactive features

## 🧪 Testing
- **Backward compatibility:** Existing code works unchanged
- **Edge cases:** Handles empty selections and limits properly
- **Cross-browser:** Consistent behavior across browsers

## 💡 Use Cases
- **Space-constrained layouts:** Limit visible tags in narrow containers
- **Clean UI:** Prevent overwhelming users with many tags
- **Mobile-friendly:** Better experience on smaller screens
- **Professional forms:** Clean appearance in business applications

## 🔄 Migration
No breaking changes - existing code continues to work without modification. The new feature is opt-in through the `maxDisplayTags` configuration option.

---

**Example Usage:**
```javascript
// Show only 1 tag + "+ X more" for additional selections
var tagSelector = MultiSelectTag('countries', {
    maxDisplayTags: 1
});

// Show 3 tags + "+ X more" for additional selections  
var tagSelector = MultiSelectTag('countries', {
    maxDisplayTags: 3
});
```

This feature significantly improves the user experience when dealing with multiple selections while maintaining the library's simplicity and flexibility.

<img width="991" height="347" alt="Screenshot 2025-07-29 at 10 25 49 AM" src="https://github.com/user-attachments/assets/bb1e30ea-6f6e-4284-8418-7c605650aec7" />
<img width="991" height="434" alt="Screenshot 2025-07-29 at 10 25 57 AM" src="https://github.com/user-attachments/assets/0be0c94e-76a5-4d67-8770-1e32f6d8951c" />

sanju9645 committed:
feat: add maxDisplayTags with interactive dropdown management

New Features:
- Add maxDisplayTags configuration option to limit visible tags
- Implement clickable '+ X more' indicator for overflow tags
- Add interactive dropdown to view and manage all selected tags
- Individual tag removal from the overflow dropdown
- Automatic dropdown hiding when no tags remain

UI/UX Improvements:
- Consistent styling with existing dropdown design
- Hover effects for better user interaction
- Smooth transitions and visual feedback
- Professional gray styling for '+ X more' indicator

Technical Implementation:
- New selectedTagsDropdown element and functionality
- Enhanced renderSelectedTags with display limit logic
- showSelectedTagsDropdown function for tag management
- Updated event handling for dropdown interactions
- Comprehensive CSS styling for new components

Documentation:
- Updated README with detailed configuration examples
- Added interactive features documentation
- Clear usage instructions and examples

This feature solves the UI clutter problem when many tags are selected
by providing a clean, interactive way to manage overflow tags.

habibmhamadi commented:
@sanju9645 Great feature. But we need the following improvements:
- The selected items in your dropdown should be unselected by clicking anywhere on the item instead of the x button.
- Remove the background color hover on the x button.
- When you unselect an item the dropdown replaces with the existing dropdown which shows unselected items. it should either remain there for unselecting more items or closes without opening the unselected items dropdown. remember when the last items is unselected the dropdown should be closed anyway.

sanju9645 committed:
feat: Make entire tag items clickable for removal

- Remove click event from X button, add to entire tag item
- Remove hover background from X button for cleaner UI
- Improve UX by making larger clickable area

sanju9645 committed:
fix: Ensure only one dropdown is visible at a time

- Hide main dropdown when showing selected items dropdown
- Hide selected items dropdown when focusing input
- Maintain clean UI with single dropdown visibility

sanju9645 commented:
I have made the changes as you suggested; please have a look.

sanju9645 commented:
@habibmhamadi  Should I make any more changes?
